replicator/src/core.rs

```diff
@@ -64,6 +69,71 @@ pub async fn start_replicator() -> anyhow::Result<()> {
     Ok(())
 }
 
+fn log_config(config: &ReplicatorConfig) {
+    log_destination_config(&config.destination);
+    log_pipeline_config(&config.pipeline);
+}
+
+fn log_destination_config(config: &DestinationConfig) {
+    match config {
+        DestinationConfig::Memory => {
+            info!("Memory config");
+        }
+        DestinationConfig::BigQuery {
+            project_id,
+            dataset_id,
+            service_account_key: _,
+            max_staleness_mins,
+        } => {
+            info!(
+                project_id,
+                dataset_id, max_staleness_mins, "BigQuery config"
+            )
+        }
+    }
+}
+
+fn log_pipeline_config(config: &PipelineConfig) {
+    info!(
+        pipeline_id = config.id,
+        publication_name = config.publication_name,
+        max_table_sync_workers = config.max_table_sync_workers,
+        "Pipeline config"
+    );
+    log_pg_connection_config(&config.pg_connection);
+    log_batch_config(&config.batch);
+    log_apply_worker_init_retry(&config.apply_worker_init_retry);
+}
+
+fn log_pg_connection_config(config: &PgConnectionConfig) {
+    info!(
+        host = config.host,
+        port = config.port,
+        dbname = config.name,
+        username = config.username,
+        tls_enabled = config.tls.enabled,
+        "Source Postgres connection config",
+    );
+}
+
+fn log_batch_config(config: &BatchConfig) {
+    info!(
+        max_size = config.max_size,
+        max_fill_ms = config.max_fill_ms,
+        "Batch config"
+    );
+}
+
+fn log_apply_worker_init_retry(config: &RetryConfig) {
+    info!(
+        max_attempts = config.max_attempts,
+        initial_delay_ms = config.initial_delay_ms,
+        max_delay_ms = config.max_delay_ms,
+        backoff_factor = config.backoff_factor,
+        "Apply worker init retry config"
+    )
+}
+
```

Comment:
Explicitly logging config instead of a debug version of `ReplicatorConfig` so that fields are captured separately in json logs in prod.